### PR TITLE
DAT-2393 Remove "dead" urls from db and from data source before insert

### PR DESCRIPTION
**Pull request by wsuchy**

### CODE REVIEW

File: extensions/wikia/HubRssFeed/models/BaseRssModel.class.php [hunk 1]
```diff
@@ -532,4 +545,43 @@ protected function findAvailableTimestamp( $timestamp, $uniqueTimestampsCount )
...
+	}
+
+	/**
+	 * Removes "dead" urls from DB
+	 * @param $feedName
+	 */
+	protected function cleanDeadUrlsInDB( $feedName ) {
+		$urlsMap = $this->getFeedData();
+		foreach ( $urlsMap as $url => $item ) {
+			if ( $item[ 'wikia_id' ] && $item[ 'page_id' ] && !$this->checkURLExists( $url ) ) {
```
Comment (SebastianMarzjan) [inline extensions/wikia/HubRssFeed/models/BaseRssModel.class.php]: Can GlobalTitle be used here? 
Check over HTTP has performance and reliability consequences

File: extensions/wikia/HubRssFeed/models/BaseRssModel.class.php [hunk 2]
```diff
@@ -532,4 +545,43 @@ protected function findAvailableTimestamp( $timestamp, $uniqueTimestampsCount )
...
+			}
+		}
+	}
+
+	/**
+	 * Removes "dead" urls from array (keys)
+	 * @param $items
+	 * @return array
+	 */
+	protected function cleanDeadUrlsFromArray( $items ) {
```
Comment (SebastianMarzjan) [inline extensions/wikia/HubRssFeed/models/BaseRssModel.class.php]: What is the difference between this and cleanDeadUrlsInDB (what is the rationale why we need both)

File: extensions/wikia/HubRssFeed/models/BaseRssModel.class.php [hunk 3]
```diff
@@ -532,4 +545,43 @@ protected function findAvailableTimestamp( $timestamp, $uniqueTimestampsCount )
...
 	}
 
+	/**
+	 * Checks if response code is different than 404. WE don't care about 503 etc as they might change
+	 * @param $url
+	 * @return bool
+	 */
+	protected function checkURLExists( $url ) {
+		$req = Http::request( "HEAD", $url, [ 'returnInstance' => true, 'followRedirects' => true ] );
+		return $req->getStatus() != "404";
```
Comment (SebastianMarzjan) [inline extensions/wikia/HubRssFeed/models/BaseRssModel.class.php]: Magic string

**Comment by jacekjursza** [comment]
LGTM

**Pull request merged**